isushibsiteaccesssync.module

```diff
@@ -0,0 +1,218 @@
+      $username = substr($username, 0, stripos($username, ','));
+
+      isushibsiteaccesssync_handle_user($username);
+    }
+
+  }
+
+  isushibsiteaccesssync_revoke_users();
+
+  db_update('isushibsiteaccess_users')
+      ->fields(array('status' => 'notAD'))
+      ->condition('status', 'Delete', '=')
+      ->execute();
+
+  ldap_close($directory_service);
+  variable_set('isushibsiteaccesssync_day_of_year', date('z') + 1);
+  watchdog('isushibsiteaccesssync', 'Finished syncing users from AD group(s).');
+}
+
+/**
+ * Mark all AD users to delete.
+ */
+function isushibsiteaccesssync_mark_users_to_delete() {
+  db_update('isushibsiteaccess_users')
+    ->fields(array('status' => 'Delete'))
+    ->condition('status', 'AD', '=')
+    ->execute();
+}
+
+/**
+ * Make sure the user has access
+ */
+function isushibsiteaccesssync_handle_user($username) {
+  // Select the user's record
+  $query = db_select('isushibsiteaccess_users', 'pu')
+    ->fields('pu', array('fuid', 'name', 'status'))
+    ->condition('name', $username, '=');
+  $results = $query->execute();
+
+  // Check if the user exists
+  if ($results->rowCount() == 1) {
+    //Update the status
+    $num_updated = db_update('isushibsiteaccess_users')
+      ->fields(array('status' => 'AD'))
+      ->condition('name', $username, '=')
+      ->execute();
+  }
+  else {
+    // Add user record.
+    $record = array(
+      'name' => $username,
+      'status' => 'AD',
+    );
+    drupal_write_record('isushibsiteaccess_users', $record);
+
+    watchdog('isushibsiteaccesssync', 'Added user %name', array('%name' => $username));
+  }
+}
+
+/**
+ * Revoke access to anyone who's status is still Delete
+ */
+function isushibsiteaccesssync_revoke_users() {
+  // Select the records where status = Delete
+  $query = db_select('isushibsiteaccess_users', 'pu')
+    ->fields('pu', array('fuid', 'name', 'status'))
+    ->condition('status', 'Delete', '=');
+  $results = $query->execute();
+
+  foreach ($results as $row) {
+    $u = user_load_by_name($row->name);
+
+    if ($u) {
+      // User exists, block the user account
+      if ($u->status != 0) {
+        $u->status = 0;
+        user_save($u);
+
+        watchdog('isushibsiteaccesssync', 'Blocked current user %name.', array('%name' => $row->name));
+      }
+    }
+    else {
+      // User doesn't exist, delete any roles from isushibsiteaccess
+      db_delete('isushibsiteaccess_roles')
+        ->condition('fuid', $row->fuid, '=')
+        ->execute();
+      watchdog('isushibsiteaccesssync', 'Deleted roles for %name.', array('%name' => $row->name));
+
+      // Delete the entry from isushibsiteaccess_users
+      db_delete('isushibsiteaccess_users')
+        ->condition('fuid', $row->fuid, '=')
+        ->execute();
+
+      watchdog('isushibsiteaccesssync', 'Deleted user entry for %name.', array('%name' => $row->name));
+    }
+  }
+}
+
+/**
+ * Return a binary key for encryption
+ */
+function isushibsiteaccesssync_get_encryption_key() {
```

Comment:
This function isn't called from anywhere. Plus it includes a hard-coded value. I assume this was done during development and isn't needed anymore.